PR for #25: Installation

nateschor opened the pull request:
In #25 we addressed an [error](https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1475045327) that was preventing installation of the package.

* git lfs was the [source](https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1476979464) of the error. We discovered this by changing LazyData to `false` [here](https://github.com/JMSLab/eventstudyr/blob/208e5f78d27647ffafbec844a535062f5efaca6e/DESCRIPTION#L49), which allowed users to download the package (although it did not contain the example data)
  * more information about the [DESCRIPTION](https://r-pkgs.org/description.html#other-fields) file
  * we removed git lfs for the datasets saved in `/data/`
* We switched from saving the example data in `.RData` format created by [build.R](https://github.com/JMSLab/eventstudyr/blob/208e5f78d27647ffafbec844a535062f5efaca6e/examples/source/exampledata/build.R) to `.rda` format created by the scripts in [/data-raw/](https://github.com/JMSLab/eventstudyr/tree/208e5f78d27647ffafbec844a535062f5efaca6e/data-raw). See [here](https://r-pkgs.org/data.html#sec-data-data-raw) for more details

Open threads:

- [x] Adding a `.gitattributes` file inside of `/data/` to [localize](https://github.com/JMSLab/eventstudyr/commit/e0357641c1f608edfdb2f805f7367cc8221c3e26#r105413008) the filtering (Nathan)
- [x] Removing the [auto-generated comment](https://github.com/JMSLab/eventstudyr/blob/208e5f78d27647ffafbec844a535062f5efaca6e/data-raw/df_EventStudyOLS_example.R#L1) at the top of each script in `/data-raw/` (Nathan, https://github.com/JMSLab/eventstudyr/pull/29/commits/8a777173f76b8f704557f96166d175a4655b343d)
- [x] Confirming that the datasets in the installation folder are now copied correctly, see [here](https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1477230756) (Santiago)
- [ ] Add a new release version, 1.0.1 (?) since this is a patch to address a bug (Santiago)

The current code for installing the package is:
```r
devtools::install_github("JMSLab/eventstudyr", ref = "issue25_installation")
``` 

@jmshapir @santiagohermo @SimonFreyaldenhoven please let me know if I've missed anything, thank you!

santiagohermo reviewed (approved):
Thanks @nateschor! 

On this:

> Confirming that the datasets in the installation folder are now copied correctly, see https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1477230756 (Santiago)

After reinstalling the package with the latest changes, the `/data/` folder looks like this

![image](https://user-images.githubusercontent.com/45404755/226711893-45d090b8-9208-4d49-87c4-80975f4806c9.png)

This is exactly the same pattern that I find in, for instance, `ggplot2`.

Furthermore, I can access the datasets with, e.g., `data(df_sample_dynamic)`, and the examples seem to work fine.

We are good on my side, so I'm approving!

--- 

Quick comment on releasing a new version. I think that `devtools::install_github("JMSLab/eventstudyr")` will always install whatever is in the main branch, so it's not strictly necessary. Still, this is an important fix so launching `1.0.1` sounds like a good idea to me.

nateschor commented:
In https://github.com/JMSLab/eventstudyr/pull/29/commits/1fc18e7283c422beb24ff655532b95a6e5bcc69a I added a `.gitattributes` file to ignore `.rda` only in `/data/`. From [here](https://github.com/git-lfs/git-lfs/issues/2120#issuecomment-290754993) it looks like the `-attr` syntax should work.

I did an experiment where I switched to [main](https://github.com/JMSLab/eventstudyr/tree/c873927f2aea2f156071f275e93864ae969ed99a/data) to see which files we were tracking with LFS (see [here](https://www.git-tower.com/learn/git/ebook/en/command-line/advanced-topics/git-lfs#:~:text=At%20some%20point%2C%20you%20might,in%20the%20current%20working%20copy.) for more info) and see how that compared with what we are tracking in [issue25_installation](https://github.com/JMSLab/eventstudyr/tree/1fc18e7283c422beb24ff655532b95a6e5bcc69a/data)

You can see below that we were picking up the old `.RData` files with lfs on main, but we're not getting the `.rda` files with how we have things currently setup. Furthermore, I reran 
```r 
devtools::install_github("JMSLab/eventstudyr", ref = "issue25_installation")
``` 
and I was able to install the package with this new `.gitattributes` setup. 

**On main**

```shell
$ git lfs ls-files
73dcb026ec - data/df_EventStudyFHS_example.RData
031b5b769f - data/df_EventStudyOLS_example.RData
298f8a7dae - data/df_sample_dynamic.RData
d42e65f17c - examples/source/raw/eventstudy_illustration_data/orig/simulation_data_dynamic.dta
c19a432e3e - examples/source/raw/eventstudy_illustration_data/orig/simulation_data_static.dta
``` 

**One issue25_installation**

```shell
$ git lfs ls-files
d42e65f17c - examples/source/raw/eventstudy_illustration_data/orig/simulation_data_dynamic.dta
c19a432e3e - examples/source/raw/eventstudy_illustration_data/orig/simulation_data_static.dta
``` 

@jmshapir does this address your concern from [here](https://github.com/JMSLab/eventstudyr/commit/e0357641c1f608edfdb2f805f7367cc8221c3e26#r105413008)?

@santiagohermo fyi

SimonFreyaldenhoven commented on data-raw/df_EventStudyFHS_example.R:
@nateschor @santiagohermo  this entire folder seems to be new, is there a reason we didn't need these function before, but now do?

nateschor replied:
@SimonFreyaldenhoven we technically don't _need_ the 4 R scripts in `data-raw` now. 

The reason I added them is because one of the things I tried on our quest to solve the [install error](https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1475045327) was looking at the data section of the [R Packages](https://r-pkgs.org/data.html) book more closely. This [section](https://r-pkgs.org/data.html#sec-data-data-raw) recommended creating a separate R scripts in a folder called `data-raw` that use the same name as the dataset they are creating. 

This is the case, for example, in the [ggplot](https://github.com/tidyverse/ggplot2/tree/962fb743281618da1e85f7692fbf8a0c9dcc9a38) repo. While this approach did not solve our error, I thought it was a better approach than what we were previously doing--creating all of the example data at the same time in a script called [build.R](https://github.com/JMSLab/eventstudyr/blob/208e5f78d27647ffafbec844a535062f5efaca6e/examples/source/exampledata/build.R) located in `examples/source/exampledata/build.R`. I can imagine the before way with `build.R` causing problems down the line with getting ready for CRAN in #28 since that is a non-standard way to include the scripts that create package example data.

Does that help?

SimonFreyaldenhoven replied:
@nateschor thanks! So right now these four functions are never evaluated, but we plan to replace the four lines in `build.R` that create the four datasets under `/data`, with the four scripts we added here prior to submitting to CRAN?

jmshapir commented on DESCRIPTION:
```diff
@@ -53,4 +53,4 @@ Suggests:
 Config/testthat/edition: 3
 Depends: 
-    R (>= 2.10)
+    R (>= 3.5.0)
```
FYI that I [removed this to-do from the roadmap](https://github.com/JMSLab/eventstudyr/wiki/Roadmap/_compare/f38e9ef7bb955eb57a5182dff03e9d97c465b3ce...decc4f7c70eecd1c8427adf8bfbe9966f62c35f5)

jmshapir commented on data/.gitattributes:
```diff
@@ -0,0 +1 @@
+*.rda -filter=lfs -diff=lfs -merge=lfs -text
```
@nateschor nice!

nateschor commented:
> Quick comment on releasing a new version. I think that devtools::install_github("JMSLab/eventstudyr") will always install whatever is in the main branch, so it's not strictly necessary. Still, this is an important fix so launching 1.0.1 sounds like a good idea to me.

@santiagohermo @jmshapir  do we need to update the release to 1.0.1 in the `issue25_installation` branch before we merge back in to main? Or is there only one release version for the entire repo, so it doesn't matter which branch we are on when we change the release number? I'm trying to be careful about making changes directly on `main`

I looked in the [GitHub docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases#about-releases) and at [release management](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) but didn't see anything that answered my question

Thanks!

santiagohermo commented:
> @santiagohermo @jmshapir do we need to update the release to 1.0.1 in the `issue25_installation` branch before we merge back in to main? Or is there only one release version for the entire repo, so it doesn't matter which branch we are on when we change the release number? I'm trying to be careful about making changes directly on `main`
> 
> I looked in the [GitHub docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases#about-releases) and at [release management](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) but didn't see anything that answered my question

Good question @nateschor! My understanding is that, when you create a release and a tag, it will point to the state of the `main` branch at that moment. Hence we should merge this repo before making the release. Two notes:

1. To maintain the structure of the first release the tag should be `v1.0.1`
2. I think we should probably follow the structure in `xtevent`, where each new releases contains a summary of changes on top, a list of PRs under a "What's changed" title, and a full changelog in the bottom. (For example, see [xtevent 2.1.1](https://github.com/JMSLab/xtevent/releases/tag/v2.1.1).)
3. As in `xtevent`, we can also consider adding a zipped file with the repo. I think that the file we would want to upload is the one created with `devtools::build()`

fyi @jmshapir

nateschor reviewed (commented):
I confirmed that I can still install `eventstudy` after the changes I just made in https://github.com/JMSLab/eventstudyr/commit/b4dfebdb84a6d808712dcfe73f01da8feb30293d and https://github.com/JMSLab/eventstudyr/commit/c09d7b94be652e9d503860ba0baac816f59e5a87, proceeding with the merge!

santiagohermo commented:
> Summary [here](https://github.com/JMSLab/eventstudyr/issues/25#issuecomment-1479725788)

Thanks @nateschor! I added a reference to #26 in the summary comment.
